src/main/java/xyz/nucleoid/plasmid/game/map/template/StagingMapTemplate.java

```diff
@@ -105,13 +117,57 @@ public MapTemplate compile() {
             }
         }
 
+        this.compileEntities(map, EntityType.ARMOR_STAND);
+        this.compileEntities(map, EntityType.ITEM_FRAME);
+        this.compileEntities(map, EntityType.PAINTING);
+
         return map;
     }
 
+    /**
+     * Compiles the entities of the specified type which are inside the map.
+     *
+     * @param map The map.
+     * @param type The entity type.
+     * @param <T> The entity type.
+     */
+    private <T extends Entity> void compileEntities(MapTemplate map, EntityType<T> type) {
+        this.world.getEntitiesByType(type, entity -> this.bounds.contains(entity.getBlockPos()))
+                .forEach(entity -> {
+                    final Vec3d oldPos = entity.getPos();
+                    final Vec3d localPos = this.globalToLocal(oldPos);
+                    BlockPos oldBlockPos = null;
+                    entity.setPos(localPos.getX(), localPos.getY(), localPos.getZ());
+
+                    if (entity instanceof AbstractDecorationEntity) {
```

Comment:
It might make sense to handle all this position-offsetting logic in `addEntity` itself somehow, since positions already have to be tweaked around there.

Reply:
Problem is, in `addEntity` it doesn't have any data to get local positions from the global context.

Reply:
Could pass the 'correct' position in and it'd just use that instead of the actual position